### DISCUSSION TITLE
`Make test` errors on fresh checkout

### DISCUSSION THREAD

**Discussion by ghost** [bluesky-social/atproto#3040] [Dev Debugging]
Hello, this test below fails every time I run `make test`. I've been trying quite a few things:

- reducing the 3000 parameter on line 1209, 
- increasing the ulimit -s value, 
- setting v8.setFlagsFromString('--stack-size=#####');

But nothing seems to help. If I lower the 3000 value too much, then the string seems to be too short, and it properly makes a round trip without 400 error and the test fails for a different reason: it expects the server to reject the too-long string.

```
packages/pds test: FAIL PDS tests/crud.test.ts (43.036 s)
packages/pds test:   ● crud operations › compare-and-swap › writes fail on values that can't reliably transform between cbor to lex
packages/pds test:     RangeError: Maximum call stack size exceeded
packages/pds test:         at JSON.stringify (<anonymous>)
packages/pds test:       1196 |     it("writes fail on values that can't reliably transform between cbor to lex", async () => {
packages/pds test:       1197 |       const passthroughBody = (data: unknown) =>
packages/pds test:     > 1198 |         ui8ToArrayBuffer(new TextEncoder().encode(JSON.stringify(data)))
packages/pds test:            |                                                        ^
packages/pds test:       1199 |
packages/pds test:       1200 |       const result = aliceAgent.call(
packages/pds test:       1201 |         'com.atproto.repo.createRecord',
packages/pds test:       at stringify (tests/crud.test.ts:1198:56)
packages/pds test:       at Object.passthroughBody (tests/crud.test.ts:1203:9)
```

Line 1209 with the seemingly arbitrary 3000 parameter value was [recently changed](https://github.com/bluesky-social/atproto/commit/d0be052e12b4bf8f8ff69027b5ac2c0f358429a7#diff-cd546031b28886f436117058f90de61b0b90033ab0046561058b0f8af3aae72b) and the commit message itself seems to indicate node v20 support, but I get better results using v18 (although this Maximum call stack error happens either way)
```
deepObject: createDeepObject(3000),
```

So, what is everyone doing to get a succesful `make test` run?

**Comment by ghost** [comment]
A fresh Debian 12 install seems to get past the `Maximum call stack size exceeded`

Now I see a lot of these errors `error in firehose event handler: sorry, too many clients already`

I also see `EADDRINUSE: address already in use :::####`

I'm studying the github actions and trying to match it closer, for example I do get better luck using flags: `--maxStreams=1 --shard=#/8 --passWithNoTests`

Is it reasonable to expect to be able to run `make test` (without flags) on a fresh checkout and reliably pass the tests locally? I see the github actions running tests successfully, so I'm not sure what's up. I'll keep trying different things.

**Reply by OracPrime** [reply]
I'm seeing the same problem on Ubuntu 24.04 - except it's so bad it brings my system down :(

Mainly seeing the too many clients already issue.  Hard to be precise, because it crashes the system so I can see the output

**Comment by bnewbold** [comment]
This will be easier to track as an "issue" than as a "discussion", so I created one here: https://github.com/bluesky-social/atproto/issues/3148

please continue the discussion over there!
